# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/basic/change-code-write-protect.js
# Commit: 81ac1990bb44
# Full Hash: 81ac1990bb44e81a2146c84d6b226408e0df0891
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2023-06-15 21:43:34
# Description:
#   Bug 1838166 - Don't allow changing code write protection at runtime. r=nbp
#   
#   To support this we'd have to reprotect all of the allocated memory pages.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180923
# ==============================================================================

diff -r e31965aaf77a -r 81ac1990bb44 js/src/jit-test/tests/basic/change-code-write-protect.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/basic/change-code-write-protect.js	Thu Jun 15 16:31:36 2023 +0000
@@ -0,0 +1,3 @@
+// |jit-test| error:Can't change code write protection at runtime
+var value = getJitCompilerOptions()["write-protect-code"];
+setJitCompilerOption("write-protect-code", Number(!value));